datafusion/src/datasource/file_format/parquet.rs

```diff
@@ -257,15 +256,15 @@ fn summarize_min_max(
 
 /// Read and parse the schema of the Parquet file at location `path`
 fn fetch_schema(object_reader: Arc<dyn ObjectReader>) -> Result<Schema> {
-    let reader = std::io::BufReader::new(object_reader.sync_reader())?;
-    let meta_data = read_metadata(&mut std::io::BufReader::new(reader))?;
+    let mut reader = object_reader.sync_reader()?;
```

Comment:
I remove this buffer now since LocalFileReader already has a buffer inside `sync_chunk_reader`, thoughts?

Reply:
good call